ios/engine/KMEI/KeymanEngineTests/FileManagementTests.swift

```diff
@@ -69,4 +69,62 @@ class FileManagementTests: XCTestCase {
       }
     }
   }
+
+  func testInstallKeyboardFromPackage() throws {
+    ResourceFileManager.shared.prepareKMPInstall(from: TestUtils.Keyboards.khmerAngkorKMP) { kmp, error in
+      XCTAssertNotNil(kmp, "Failed to prepare KMP for installation")
+      XCTAssertNil(error, "Error occurred while preparing KMP for installation")
+      XCTAssertNotNil(kmp as? KeyboardKeymanPackage, "KMP resource type improperly recognized - expected a keyboard package!")
+
+      do {
+        try ResourceFileManager.shared.install(TestUtils.Keyboards.khmer_angkor, from: kmp!)
+      } catch {
+        XCTFail("Unexpected error during KeyboardPackage install")
+      }
+
+      let installURL = Storage.active.keyboardURL(forID: "khmer_angkor", version: "1.0.6")
+
+      XCTAssertTrue(FileManager.default.fileExists(atPath: installURL.path),
+                    "Could not find installed keyboard file")
+
+      let keyboards = Storage.active.userDefaults.userKeyboards!
+
+      XCTAssertEqual(keyboards.count, 1, "Unexpected number of keyboards were installed")
+      XCTAssertEqual(keyboards[0].id, "khmer_angkor", "Installed keyboard ID mismatch")
+
+      // While the KMP's version of the specified InstallableKeyboard does specify Fonts,
+      // the literal-based testing version does NOT.  Since we're installing from the predefined,
+      // test-copy instance, we expect NOT to see the font from this test's install!
+      //
+      // Yes, the KeymanPackage.contains check isn't exhaustive - it just does a pair of
+      // id checks.  This version of events couldn't happen if we thorough enough there.
+      let fontURL = Storage.active.fontURL(forKeyboardID: "khmer_angkor", filename: "Mondulkiri-R.ttf")
+      XCTAssertFalse(FileManager.default.fileExists(atPath: fontURL.path))
```

Comment:
For reference, this is the unit test I was referring to in the `contains` review comment.